### ISSUE TITLE
Settings > System & Accounts > Users

### ISSUE THREAD

**Issue by maria-komarova** [pop-os/cosmic-settings#443]
Users panel provides a way to change user settings and add/remove new users. 

![Users-default](https://github.com/user-attachments/assets/7fce3a8f-3336-43cf-9c18-2c795411b177)

State when another user has been added:
![Users-two-users](https://github.com/user-attachments/assets/d0de7d37-dbac-4a88-871a-bd7456070150)

[Figma file with more details](https://www.figma.com/design/Z6Z9Ox6WNkiiM8gcAkcElS/COSMIC-Settings-First-Release---Developer-Handoff?node-id=13918-2014&t=cagN4rG2n71Xfuj0-1).

**Comment by maria-komarova** [comment]
The settings for the page are not implemented yet. They are on the roadmap for alpha 2 release.

**Comment by Psyhackological** [comment]
I see. Can I change my account image in other way in alpha 1?

**Comment by maria-komarova** [comment]
> I see. Can I change my account image in other way in alpha 1?

I don't know about one right now. 

> it's support for fingerprint/biometrics/pin/passkey planned for user sign in? I see just password accordance in the figma mock-up

This will be out of scope for the first release but we'll definitely be looking into it in the future.

**Comment by jotuel** [comment]
> I see. Can I change my account image in other way in alpha 1?

You can change the users icon by (as root) overwriting the file /var/lib/AccountsService/icons/{user}. 


> it's support for fingerprint/biometrics/pin/passkey planned for user sign in? I see just password accordance in the figma mock-up

Fingerprint is already supported but setting it currently is up to you as a user.  (As root) `apt install libpam-fprintd && fprintd-enroll && pam-auth-update`.

**Comment by Psyhackological** [comment]
> > I see. Can I change my account image in other way in alpha 1?
> 
> You can change the users icon by (as root) overwriting the file /var/lib/AccountsService/icons/{user}. 
> 

Thank you so much!

**Comment by mmstick** [comment]
@Psyhackological No need to use root to change the icon. The icon path is by default defined to point to `~/.face`, as defined by `Icon=` in `/var/lib/AccountsService/users/{{user}}`.

**Comment by jotuel** [comment]
Cool. :) Mine looks like `root@pop-os:~# cat /var/lib/AccountsService/users/jwnz
[User]
Icon=/var/lib/AccountsService/icons/jwnz
SystemAccount=false` which explains why a copy of the same photo at $HOME/.face didn't and doesn't work for me.

**Comment by acolombier** [comment]
Hi, I have started implemented that feature, but a few questions arise from the initial work:
- Only custom avatars appears to be designed, but the system should contain a list of predefined avatar in `/usr/share/pixmaps/faces/`. Is this coming in phase two? (Can't find a link to the Figma file)
- Authentication is suggested only for password and role update. IMHO, allowing username change is a privileged operation as it can cause various security problems. Shall we simply require authentication for any user changes?
- The DBus interface doesn't seem to require any kind of authentication. Does that mean that the locking is only performed on the frontend?
- The authentication popup appears to be a static password authentication, instead of using PAM, which could result in alternative authentication (in my case, fingerprint). Do we expect that lack to be addressed on phase two as well?
- What to do for disabled user and disabling/enabling operation?

Some more technical help - maybe @mmstick you can help again?
- Is there any widget allowing to clip an image into a a shape/rounded rect/circle already or shall I create a component for it?
- Can a symbolic icon be colored with accent color? I'm pretty sure it's possible but I couldn't find an example/doc or make sense of the code.

Here is the current state of progress:

https://github.com/user-attachments/assets/3cc5a4a3-21d4-4bd5-b9c2-2ca2caceec2d

**Comment by maria-komarova** [comment]
> Only custom avatars appears to be designed, but the system should contain a list of predefined avatar in /usr/share/pixmaps/faces/. Is this coming in phase two? (Can't find a link to the Figma file)

Yes, that wasn't planned for first release, but for the second. 

> The authentication popup appears to be a static password authentication, instead of using PAM, which could result in alternative authentication (in my case, fingerprint). Do we expect that lack to be addressed on phase two as well?

Again, fingerprint was postponed till future releases. 

> Authentication is suggested only for password and role update. IMHO, allowing username change is a privileged operation as it can cause various security problems. Shall we simply require authentication for any user changes?

What kinds of security problems?

> What to do for disabled user and disabling/enabling operation?

Could you go into more detail on the situations when we'd have disabling or disabled users?

**Comment by acolombier** [comment]
> What kinds of security problems?

The main concern is deny of service: system authentication requires a valid couple username/password, if an attacker is able to randomize any of these two, then they could prevent access to the machine. This is particularly worrying for user like me, that may have SSH/RDP enabled on their machine so they can acces their machine when they are on the go.

> Could you go into more detail on the situations when we'd have disabling or disabled users?

User account may be locked ([doc](https://linux.die.net/man/8/usermod), `-L/-U`). I appreciate we may not want to include this operational support on phase one, but I'm wondering how to proceed with currently disabled user? Shall we hide them for the list for now? Show them greyed out?

**Comment by acolombier** [comment]
> But we are not talking about changing a users login here though are we?

It appears this is currently the spec

![image](https://github.com/user-attachments/assets/e5184001-bee0-4362-a442-f2fd417ee397)

> So the assumption likely is that these are always kept hidden?

I don't quite like the idea of obfuscating the information to the user. Hiding system account definitely makes sense, for the stability reason you've mentioned. Not implementing every single feature of `usermod` also make sense, but the ability to lock/unlock account is a behaviour that will be useful to non-technical user. 

As someone that have been using Linux for the last 15 years, I really would like to see distros becoming more inclusive, and stop segregating users between nerd like us that knows UNIX command line tools, and other :) 

In the meantime, I fully endorse the concept of iteration and priority, so if we were to say that it would come later (like fingerprint), this would be completely fair.

**Comment by jotuel** [comment]
```
jwnz@pop-os:~$ finger jwnz | head -n 1 && finger root | head -n 1
Login: jwnz           			Name: Joonas Tuomi
Login: root           			Name: root
```
We could presume users name and username are different things. 

On the second part I hear you. I also believe that if you can muck things up in a GUI you should only ever need GUI to fix your mistakes. Of course this is solvable right: one would only need to always keep track that at least one admin account remains unlocked (and that its password is still known). Then again this is just the thing I came up in a couple of minutes of thinking for an example where an end user would end up in a bad spot. It would just require some testing. Just tried and you can run the command while logged in as the "only user", ill post this before logging out myself. 

> (like fingerprint)

Well that static password popup most likely is pam and supports fingerprint (depending on your `/etc/pam.d/sudo`), or at least I am able to just pop my finger in and the pop up disappears. Up to the user to set it up for now.

**Comment by dragon99919** [comment]
Is there "autologin without password" option planed? Or is it something planned in display manager?

**Comment by maria-komarova** [comment]
Planned for after the first release.

**Comment by UMineiro** [comment]
Will there be an option to configure group memberships? Please do not go the Gnome e KDE way of removing this functionality.

**Comment by mmstick** [comment]
@acolombier Do you have a branch that you'd like to submit? We are ready to work on this now.

**Comment by UMineiro** [comment]
Unfortunately no. Not a Rust developer, yet.

Em qui., 7 de nov. de 2024 às 13:10, Michael Murphy <
***@***.***> escreveu:

**Comment by acolombier** [comment]
I do @mmstick - https://github.com/acolombier/cosmic-settings/tree/feat/user-accounts

I just broke it as I tried to rebased it. I'll try to push a fix in the next few hours.

It's only frontend tho, as I didn't know much of how to interact with D-Bus and authentication. I'd be keen to continue working on this tho, in case there is some bits you would be okay to let me help with?
